src/main/java/seedu/address/logic/commands/EditNoteCommand.java

```diff
@@ -0,0 +1,226 @@
+import java.util.Optional;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.note.Description;
+import seedu.address.model.person.note.Note;
+
+/**
+ * Edits an appointment in the address book.
+ */
+public class EditNoteCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit-an";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the appointment note identified "
+            + "by the patient index number used in the displayed patient list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: PATIENT_INDEX (must be a positive integer) "
+            + "NOTE_INDEX (must be a positive integer) "
+            + "[" + PREFIX_DATE + "DATE<DD-MM-YYYY>] "
+            + "[" + PREFIX_TIME + "TIME<HHMM>] "
+            + "[" + PREFIX_NOTE + "NOTE] "
+            + "Example: " + COMMAND_WORD + " 1 1 "
+            + PREFIX_DATE + "19-02-2024 "
+            + PREFIX_TIME + "1430 "
+            + PREFIX_NOTE + "General Flu ";
+
+    public static final String MESSAGE_EDIT_NOTE_SUCCESS = "Edited Note: %1$s";
+    public static final String MESSAGE_NOTE_NOT_EDITED = "No fields edited. Please try again.";
+    private final Index patientIndex;
+    private final Index noteIndex;
+    private final EditNoteDescriptor editNoteDescriptor;
+
+    /**
+     * @param patientIndex         of the patient with the appointment note to edit
+     * @param noteIndex            of the note in the filtered note list to edit
+     * @param editNoteDescriptor   details to edit the note with
+     */
+    public EditNoteCommand(Index patientIndex, Index noteIndex, EditNoteDescriptor editNoteDescriptor) {
+        requireNonNull(patientIndex);
+        requireNonNull(noteIndex);
+        requireNonNull(editNoteDescriptor);
+
+        this.patientIndex = patientIndex;
+        this.noteIndex = noteIndex;
+        this.editNoteDescriptor = editNoteDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownPatientList = model.getFilteredPersonList();
+
+        if (patientIndex.getZeroBased() >= lastShownPatientList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+        } else {
+            if (noteIndex.getZeroBased() >= lastShownPatientList.get(patientIndex.getZeroBased()).getNotes().size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_INDEX);
+            }
+        }
```

Comment:
Nitpick, but I think we can avoid the additional nesting by place the `if` statement on the top level.